src/lib/layouts/workflow-history-layout.svelte

```diff
@@ -17,48 +21,73 @@
   import InputAndResults from '$lib/components/workflow/input-and-results.svelte';
   import WorkflowDetail from '$lib/components/workflow/workflow-detail.svelte';
   import Accordion from '$lib/holocene/accordion.svelte';
-  import { timeFormat } from '$lib/stores/time-format';
-  import { exportHistory } from '$lib/utilities/export-history';
-  import { getWorkflowStartedCompletedAndTaskFailedEvents } from '$lib/utilities/get-started-completed-and-task-failed-events';
   import ChildWorkflowsTable from '$lib/components/workflow/child-workflows-table.svelte';
-  import Button from '$lib/holocene/button.svelte';
   import EventShortcutKeys from '$lib/components/event/event-shortcut-keys.svelte';
 
   let showShortcuts = false;
 
+  $: namespace = $page.params.namespace;
+  $: workflowId = $page.params.workflow;
+  $: runId = $page.params.run;
+
+  $: $refresh, getStartAndEndEvents(namespace, workflowId, runId);
+
   $: workflowEvents =
     getWorkflowStartedCompletedAndTaskFailedEvents($eventHistory);
+
+  const getStartAndEndEvents = async (
+    namespace: string,
+    workflowId: string,
+    runId: string,
+  ) => {
+    const { settings, user } = $page.data;
+    const events = await fetchStartAndEndEvents({
+      namespace,
+      workflowId,
+      runId,
+      settings,
+      accessToken: user?.accessToken,
+    });
+    $eventHistory = events;
+  };
+
+  const onViewClick = (view: EventView) => {
+    if ($page.url.searchParams.get('page')) {
```

Comment:
I think we might want to do something similar for page size. Currently when you select a new page size the page remains the same and you can end up in a weird state (specifically in the `Compact` view) 🔽 

![Screenshot 2023-01-12 at 1 14 22 PM](https://user-images.githubusercontent.com/15069288/212171585-0bdc0701-549e-40dd-88c1-5e497ee48f9f.png)

or with an incorrect url in the `History` view.